src/DynamoCoreWpf/ViewModels/Core/OutPortViewModel.cs

```diff
@@ -0,0 +1,280 @@
+        /// </summary>
+        public Visibility UseContextMenuVisibility
+        {
+            get
+            {
+                if (IsPortCondensed)
+                {
+                    return Visibility.Collapsed;
+                }
+                else
+                {
+                    return Visibility.Visible;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Determines whether the output port button says 'Hide Wires' or 'Show Wires'
+        /// </summary>
+        public string ShowHideWiresButtonContent
+        {
+            get => showHideWiresButtonContent;
+            set
+            {
+                showHideWiresButtonContent = value;
+                RaisePropertyChanged(nameof(ShowHideWiresButtonContent));
+            }
+        }
+
+        /// <summary>
+        /// Sets the visibility of the connectors from the port. This will overwrite the 
+        /// individual visibility of the connectors. However when visibility is controlled 
+        /// from the connector, that connector's visibility will overwrite its previous state.
+        /// In order to overwrite visibility of all connectors associated with a port, us this 
+        /// flag again.
+        /// </summary>
+        public bool SetConnectorsVisibility
+        {
+            get => areConnectorsHidden;
+            set
+            {
+                areConnectorsHidden = value; 
+                RaisePropertyChanged(nameof(SetConnectorsVisibility));
+            }
+        }
+
+        /// <summary>
+        /// Enables or disables the Hide Wires button on the node output port context menu.
+        /// </summary>
+        public bool HideWiresButtonEnabled
+        {
+            get => hideWiresButtonEnabled;
+            set
+            {
+                hideWiresButtonEnabled = value; 
+                RaisePropertyChanged(nameof(HideWiresButtonEnabled));
+            }
+        }
+
+        /// <summary>
+        /// Takes care of the multiple UI concerns when dealing with the Unhide/Hide Wires button
+        /// on the output port's context menu.
+        /// </summary>
+        private void RefreshHideWiresButton()
+        {
+            HideWiresButtonEnabled = _port.Connectors.Count > 0;
+            SetConnectorsVisibility = CheckIfConnectorsAreHidden();
+
+            ShowHideWiresButtonContent = SetConnectorsVisibility
+                ? Wpf.Properties.Resources.ShowWiresPopupMenuItem
+                : Wpf.Properties.Resources.HideWiresPopupMenuItem;
+
+            RaisePropertyChanged(nameof(ShowHideWiresButtonContent));
+        }
+
+        #endregion
+
+        public OutPortViewModel(NodeViewModel node, PortModel port) :base(node, port)
+        {
+            _node.PropertyChanged += _node_PropertyChanged;
+            _port.PropertyChanged += _port_PropertyChanged;
+            RefreshHideWiresButton();
+        }
+
+        public override void Dispose()
```

Comment:
where does this get called? (same question for inports?)

Reply:
[Here](https://github.com/DynamoDS/Dynamo/blob/master/src/DynamoCoreWpf/ViewModels/Core/NodeViewModel.cs#L752) and [Here](https://github.com/DynamoDS/Dynamo/blob/master/src/DynamoCoreWpf/ViewModels/Core/NodeViewModel.cs#L1063). But good question.  We missed some [here](https://github.com/DynamoDS/Dynamo/blob/master/src/DynamoCoreWpf/ViewModels/Core/NodeViewModel.cs#L1098).  I will add that to this PR